reconcile/external_resources/secrets_sync.py

```diff
@@ -22,13 +26,21 @@
 )
 from reconcile.utils.oc_map import OCMap, init_oc_map_from_clusters
 from reconcile.utils.openshift_resource import OpenshiftResource, ResourceInventory
-from reconcile.utils.secret_reader import SecretReaderBase
+from reconcile.utils.secret_reader import SecretNotFound, SecretReaderBase
 from reconcile.utils.three_way_diff_strategy import three_way_diff_using_hash
 from reconcile.utils.vault import (
     VaultClient,
     _VaultClient,  # noqa
 )
 
+SECRET_ANN_PREFIX = "external-resources"
+SECRET_ANN_PROVISION_PROVIDER = SECRET_ANN_PREFIX + "/provision_provider"
+SECRET_ANN_PROVISIONER = SECRET_ANN_PREFIX + "/provisioner_name"
+SECRET_ANN_PROVIDER = SECRET_ANN_PREFIX + "/provider"
+SECRET_ANN_IDENTIFIER = SECRET_ANN_PREFIX + "/identifier"
+SECRET_UPDATED_AT = SECRET_ANN_PREFIX + "/updated_at"
+SECRET_UPDATED_AT_TIMEFORMAT = "%d/%m/%Y %H:%M:%S"
```

Comment:
btw, if you want to keep it same as our json schema `date-time` or k8s timestamp format, then this has to be a timezone aware string, `.isoformat()` will produce `20240101T00:00:00+00:00`, you can also specify `Z` explicitly as we only use UTC `%Y-%m-%dT%H:%M:%SZ`.

And because utc iso time string is comparable, we don't even need to parse it back to `datetime`, can directly do string compare in time check.

Reply:
TIL. Added.